src/python/pants/backend/python/providers/pyenv/rules_integration_test.py

```diff
@@ -45,6 +45,7 @@ def run_run_request(
     args = [
         "--backend-packages=['pants.backend.python', 'pants.backend.python.providers.experimental.pyenv']",
         "--source-root-patterns=['src']",
+        "--pyenv-python-provider-installation-extra-env-vars=['HOME']",
```

Comment:
Installing Python 2.7 via pyenv requires access to brew (to get a compatible openssl), and brew only runs if HOME is set.